crates/holochain/src/conductor/cell.rs

```diff
@@ -980,38 +780,41 @@ impl Cell {
     pub async fn call_zome(
         &self,
         call: ZomeCall,
-        workspace_lock: Option<HostFnWorkspace>,
+        workspace_lock: Option<SourceChainWorkspace>,
     ) -> CellResult<ZomeCallResult> {
         // Check if init has run if not run it
         self.check_or_run_zome_init().await?;
 
-        let arc = self.env();
-        let keystore = arc.keystore().clone();
+        let keystore = self.conductor_api.keystore().clone();
 
         // If there is no existing zome call then this is the root zome call
         let is_root_zome_call = workspace_lock.is_none();
         let workspace_lock = match workspace_lock {
             Some(l) => l,
             None => {
-                HostFnWorkspace::new(
-                    self.env().clone(),
+                SourceChainWorkspace::new(
+                    self.authored_env().clone(),
+                    self.dht_env().clone(),
                     self.cache().clone(),
+                    keystore.clone(),
                     self.id.agent_pubkey().clone(),
                 )
                 .await?
             }
         };
 
-        let conductor_api = self.conductor_api.clone();
+        let conductor_handle = self.conductor_handle.clone();
         let signal_tx = self.signal_broadcaster().await;
         let ribosome = self.get_ribosome().await?;
-        let invocation = ZomeCallInvocation::from_interface_call(conductor_api.clone(), call).await;
+        let invocation =
+            ZomeCallInvocation::from_interface_call(self.conductor_api.clone(), call).await;
 
         let args = CallZomeWorkflowArgs {
+            cell_id: self.id.clone(),
```

Comment:
note this goes away but maybe you should merge your stuff first

Reply:
H,, I wonder if this is a problem? How does the host function find out the agent without this?